extensions/blocks/donations/amount-input.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import { CURRENCIES } from '@automattic/format-currency';
+
+/**
+ * WordPress dependencies
+ */
+import { RichText } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { minimumTransactionAmountForCurrency } from '../../shared/currencies';
+
+const parseAmount = ( amount, currency ) => {
+	if ( ! amount ) {
+		return null;
+	}
+
+	amount = parseFloat(
+		amount
+			// Remove any thousand grouping separator.
+			.replace( new RegExp( '\\' + CURRENCIES[ currency ].grouping, 'g' ), '' )
+			// Replace the localized decimal separator with a dot (the standard decimal separator in float numbers).
+			.replace( new RegExp( '\\' + CURRENCIES[ currency ].decimal, 'g' ), '.' )
+	);
```

Comment:
If it's well formatted, it's fine to leave this as a string value on the client, and let the server parse. But okay to defer here too, if other pieces need heavy refactoring to do this.